src/client/options/mod.rs

```diff
@@ -675,43 +675,181 @@ impl Serialize for ClientOptions {
     }
 }
 
+/// Contains the options that can be set via a MongoDB connection string.
+///
+/// The format of a MongoDB connection string is described [here](https://docs.mongodb.com/manual/reference/connection-string/#connection-string-formats).
 #[derive(Debug, Default, PartialEq)]
```

Comment:
I think it would be nice to have a `Deserialize` implementation that parses a string. See my other comment below about going back to strings which we'd use for a `Display` and `Serialize` implementation.

Reply:
Done.